aodncore/util/misc.py

```diff
@@ -95,6 +96,22 @@ def format_exception(exception):
     return "{cls}: {message}".format(cls=exception.__class__.__name__, message=exception)
 
 
+def get_pattern_subgroups_from_string(string, pattern):
+    """Function to retrieve parts of a string given a compiled pattern (re.compile(str))
+    :return: dictionary of fields matching a given pattern
+    """
+    retype = type(re.compile(''))
+    if not isinstance(pattern, retype):
+        try:
+            pattern = re.compile(pattern)
+        except Exception as err:
+            raise TypeError("error compiling pattern '{pattern}': {e}".
+                            format(pattern=pattern, e=format_exception(err)))
+
+    m = pattern.match(string)
```

Comment:
Note that this will only return anything if the pattern matches the _beginning_ of the string (see https://docs.python.org/2/library/re.html#re.RegexObject.match). So
* there's no need to start the pattern with "^"; and
* if you want to match _anywhere_ in the string, start the pattern with ".*".

Maybe worth noting this in the doc string?